### DISCUSSION TITLE
Turning off metrics from infinispan-embedded

### DISCUSSION THREAD

**Discussion by zbhavyai** [quarkusio/quarkus#26263] [Q&A] [answered]
Hello,

I have a project where `quarkus-infinispan-embedded` is being used for failover. In the same project I would like to have some application specific metrics generated using `quarkus-micrometer-registry-prometheus`. However, just by having these two dependencies throws below error during build:

```
ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.ChainBuildException: Multiple producers of item class io.quarkus.deployment.metrics.MetricsCapabilityBuildItem (io.quarkus.smallrye.metrics.deployment.SmallRyeMetricsProcessor#metricsCapabilityBuildItem)

...

io.quarkus.deployment.metrics.MetricsCapabilityBuildItem (io.quarkus.smallrye.metrics.deployment.SmallRyeMetricsProcessor#metricsCapabilityBuildItem)
Caused by: java.lang.Throwable: This is the location of the conflicting producer (io.quarkus.micrometer.deployment.MicrometerProcessor#metricsCapabilityPrometheusBuildItem). Use -Dquarkus.builder.log-conflict-cause=true to see the full stacktrace.
```

I checked the mvn dependency tree, and found that `quarkus-infinispan-embedded` does include `quarkus-smallrye-metrics`. Is there any way to disable it, or prevent conflict of these "producers"?

I have seen the issue [24032](https://github.com/quarkusio/quarkus/issues/24032), but didn't find it helpful. Also, I have already tested `quarkus.smallrye-metrics.extensions.enabled=false`  but it doesn't work.

Any help is greatly appreciated.

Thank you.

[a comment by an automated account is not shown]

**Reply by zbhavyai** [reply]
Made a reproducer at [https://github.com/zbhavyai/quarkus-infinispan-metrics-reproducer](https://github.com/zbhavyai/quarkus-infinispan-metrics-reproducer).

**Reply by jmartisk** [reply]
The `infinispan-embedded` extension has been discontinued since Quarkus 1.7 - your reproducer seems to be using Quarkus 2.9 but importing the extension from Quarkus 1.6.1 where it was still present, this combination will quite possibly not work at all. Besides, the 1.6.1 version of the extension contains a hard dependency on the SmallRye Metrics extension which probably makes it impossible to use it with Micrometer. This would normally be considered an issue, but given the discontinuation of the extension, this won't be fixed.

AFAIK there is some effort by the Infinispan community to continue this extension in a separate repository: https://github.com/infinispan/infinispan-quarkus but I'm not sure about its status. It seems they currently don't even deploy their releases into public Maven repos. 
If you really want to use Infinispan embedded, I would suggest you try and see if this can help. 
But instead, you might want to switch to using `quarkus-cache` if you want to use an embedded engine for caching. This extension doesn't currently support metrics so there can't be any conflict. See https://quarkus.io/guides/cache

**Comment by aloubyansky** [comment, marked as answer]
There is a way around it though. Here is how you can exclude `quarkus-smallrye-metrics`:
````
[aloubyansky@localhost quarkus-infinispan-metrics-reproducer]$ git diff
diff --git a/pom.xml b/pom.xml
index ef1445f..2357266 100644
--- a/pom.xml
+++ b/pom.xml
@@ -26,6 +26,21 @@
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-infinispan-embedded</artifactId>
+        <version>1.6.1.Final</version>
+        <exclusions>
+          <exclusion>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-metrics</artifactId>
+          </exclusion>
+          <exclusion>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-metrics-deployment</artifactId>
+          </exclusion>
+        </exclusions>
+      </dependency>
     </dependencies>
   </dependencyManagement>
 
@@ -37,7 +52,6 @@
     <dependency>
       <groupId>io.quarkus</groupId>
       <artifactId>quarkus-infinispan-embedded</artifactId>
-      <version>1.6.1.Final</version>
````
It'll boot.

**Reply by zbhavyai** [reply]
Thank you so much @aloubyansky and @jmartisk ! After putting the suggested exclusions, no longer facing that issue.
